src/main/scala/is/hail/variant/GenomeReference.scala

```diff
@@ -157,16 +157,30 @@ case class GenomeReference(name: String, contigs: Array[String], lengths: Map[St
     }
   }
 
-  def unify(concrete: GenomeReference): Boolean = this == concrete
+  def unify(concrete: GRBase): Boolean = this == concrete
 
   def isBound: Boolean = true
 
   def clear() {}
 
   def subst(): GenomeReference = this
+
+  override def toString: String = name
 }
 
 object GenomeReference {
+  var references: Map[String, GenomeReference] = Map("GRCh37" -> GRCh37, "GRCh38" -> GRCh38)
+
+  def addReference(gr: GenomeReference) {
+    references += (gr.name -> gr)
```

Comment:
This probably shouldn't be able to modify references.  What happens if you load a dataset, overwrite the b37 reference, load another dataset and then join them?